### ISSUE TITLE
configure: error: GDALAllRegister not found in libgdal

### ISSUE THREAD

**Issue by mjfrigaard** [rspatial/terra#1461]
Hello! I am here because I was attempting to install [mapboxapi ](https://walker-data.com/mapboxapi/) (which has the following dependencies): 

```
also installing the dependencies ‘terra’, ‘sf’, ‘slippymath’, ‘protolite’, ‘magick’, ‘leaflet’, ‘raster’
```

The install for mapboxapi resulted in the following error (from the first terra package dependency):

```
collect2: error: ld returned 1 exit status
configure: Install failure: compilation and/or linkage problems.
configure: error: GDALAllRegister not found in libgdal.
ERROR: configuration failed for package ‘terra’
* removing ‘/home/xsn6-su/R/x86_64-redhat-linux-gnu-library/4.3/terra’
```

The GDALAllRegister error was tracked to the following information on Linux installations on the terra README: https://github.com/rspatial/terra/blob/master/README.md#linux

> See the sf [instructions](https://github.com/r-spatial/sf) for installation on other linux systems --- and for possible updates/improvements on the above instructions.

The [Linux/Fedora installation commands](https://github.com/r-spatial/sf/blob/main/README.md#fedora) are below:

```
|- # > dnf install gdal-devel proj-devel geos-devel sqlite-devel udunits2-devel
Repository packages-microsoft-com-prod is listed more than once in the configuration
Repository ol8_developer_EPEL is listed more than once in the configuration
Repository ol8_UEKR6 is listed more than once in the configuration
Last metadata expiration check: 0:03:45 ago on Tue 19 Mar 2024 04:43:28 PM UTC.
Package gdal-devel-3.0.4-11.el8.x86_64 is already installed.
Package udunits2-devel-2.2.26-5.el8.x86_64 is already installed.
Dependencies resolved.
=========================================================================================
 Package           Arch        Version                     Repository               Size
=========================================================================================
Installing:
 geos-devel        x86_64      3.7.2-1.el8                 ol8_developer_EPEL      1.4 M
 proj-devel        x86_64      6.3.2-4.el8                 ol8_developer_EPEL       90 k
 sqlite-devel      x86_64      3.26.0-19.0.1.el8_9         ol8_baseos_latest       164 k

Transaction Summary
=========================================================================================
Install  3 Packages

Total download size: 1.7 M
Installed size: 18 M
Is this ok [y/N]: y
Downloading Packages:
(1/3): sqlite-devel-3.26.0-19.0.1.el8_9.x86_64.rpm       728 kB/s | 164 kB     00:00
(2/3): geos-devel-3.7.2-1.el8.x86_64.rpm                 5.3 MB/s | 1.4 MB     00:00
(3/3): proj-devel-6.3.2-4.el8.x86_64.rpm                  30 kB/s |  90 kB     00:02
-----------------------------------------------------------------------------------------
Total                                                    583 kB/s | 1.7 MB     00:02
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                 1/1
  Installing       : sqlite-devel-3.26.0-19.0.1.el8_9.x86_64                         1/3
  Installing       : proj-devel-6.3.2-4.el8.x86_64                                   2/3
  Installing       : geos-devel-3.7.2-1.el8.x86_64                                   3/3
  Running scriptlet: geos-devel-3.7.2-1.el8.x86_64                                   3/3
  Verifying        : geos-devel-3.7.2-1.el8.x86_64                                   1/3
  Verifying        : proj-devel-6.3.2-4.el8.x86_64                                   2/3
  Verifying        : sqlite-devel-3.26.0-19.0.1.el8_9.x86_64                         3/3

Installed:
  geos-devel-3.7.2-1.el8.x86_64                    proj-devel-6.3.2-4.el8.x86_64
  sqlite-devel-3.26.0-19.0.1.el8_9.x86_64

Complete!
```

After installation, the installation for these packages was verified: 

```
|- # > dnf install gdal-devel proj-devel geos-devel sqlite-devel udunits2-devel
Repository packages-microsoft-com-prod is listed more than once in the configuration
Repository ol8_developer_EPEL is listed more than once in the configuration
Repository ol8_UEKR6 is listed more than once in the configuration
Last metadata expiration check: 0:01:02 ago on Tue 19 Mar 2024 05:53:22 PM UTC.
Package gdal-devel-3.0.4-11.el8.x86_64 is already installed.
Package proj-devel-6.3.2-4.el8.x86_64 is already installed.
Package geos-devel-3.7.2-1.el8.x86_64 is already installed.
Package sqlite-devel-3.26.0-19.0.1.el8_9.x86_64 is already installed.
Package udunits2-devel-2.2.26-5.el8.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
```

The installation of the development version of the terra package still results in the same error: 

```r
remotes::install_github("rspatial/terra")
Downloading GitHub repo rspatial/terra@HEAD
Running `R CMD build`...
* checking for file ‘/tmp/RtmpLCfSHY/remotes3fb6674ae711e/rspatial-terra-ba0a5a0/DESCRIPTION’ ... OK
* preparing ‘terra’:
* checking DESCRIPTION meta-information ... OK
* cleaning src
* running ‘cleanup’
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building ‘terra_1.7-74.tar.gz’
Installing package into ‘/home/xsn6-su/R/x86_64-redhat-linux-gnu-library/4.3’
(as ‘lib’ is unspecified)
* installing *source* package ‘terra’ ...
** using staged installation
configure: CC: gcc
configure: CXX: g++ -std=gnu++17
checking for gdal-config... /usr/bin/gdal-config
checking gdal-config usability... yes
configure: GDAL: 3.0.4
checking GDAL version >= 2.0.1... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for gdal.h... yes
checking GDAL: linking with --libs only... no
checking GDAL: linking with --libs and --dep-libs... no
/usr/lib64/libgdal.so: undefined reference to `PQcmdStatus@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQfname@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQsetNoticeProcessor@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQftable@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQstatus@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQftablecol@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQconnectdb@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQescapeStringConn@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `lo_close@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQputCopyEnd@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQgetvalue@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQgetisnull@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQftype@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQfmod@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `lo_creat@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQresultStatus@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `lo_read@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQexec@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQsetClientEncoding@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQgetlength@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQputCopyData@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQgetResult@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `lo_write@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQfinish@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQclear@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQerrorMessage@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQnfields@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `lo_open@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQexecParams@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQntuples@RHPG_9.6'
/usr/lib64/libgdal.so: undefined reference to `PQresultErrorMessage@RHPG_9.6'
collect2: error: ld returned 1 exit status
/usr/bin/ld: cannot find -larmadillo
/usr/bin/ld: cannot find -ljson-c
/usr/bin/ld: cannot find -lfreexl
/usr/bin/ld: cannot find -lwebp
/usr/bin/ld: cannot find -lkmlbase
/usr/bin/ld: cannot find -lkmldom
/usr/bin/ld: cannot find -lkmlengine
/usr/bin/ld: cannot find -lkmlxsd
/usr/bin/ld: cannot find -lkmlregionator
/usr/bin/ld: cannot find -lxerces-c
/usr/bin/ld: cannot find -lnetcdf
/usr/bin/ld: cannot find -lhdf5
/usr/bin/ld: cannot find -lmfhdf
/usr/bin/ld: cannot find -ldf
/usr/bin/ld: cannot find -logdi
/usr/bin/ld: cannot find -lgif
/usr/bin/ld: cannot find -lgeotiff
/usr/bin/ld: cannot find -ltiff
/usr/bin/ld: cannot find -lgta
/usr/bin/ld: cannot find -lcfitsio
/usr/bin/ld: cannot find -ldapclient
/usr/bin/ld: cannot find -ldapserver
/usr/bin/ld: cannot find -ldap
/usr/bin/ld: cannot find -lspatialite
collect2: error: ld returned 1 exit status
configure: Install failure: compilation and/or linkage problems.
configure: error: GDALAllRegister not found in libgdal.
ERROR: configuration failed for package ‘terra’
* removing ‘/home/xsn6-su/R/x86_64-redhat-linux-gnu-library/4.3/terra’
Warning message:
In i.p(...) :
  installation of package ‘/tmp/RtmpLCfSHY/file3fb667239559f0/terra_1.7-74.tar.gz’ had non-zero exit status
```

The issue seems to begin here: 

```
checking for gdal.h... yes
checking GDAL: linking with --libs only... no
checking GDAL: linking with --libs and --dep-libs... no
```

Can you suggest any documentation/resources for addressing the 'compilation and/or linkage problems'? 

Thank you!

**Comment by rhijmans** [comment]
Sorry for the lack of response. Have you been able to find a solution? May this be caused by having multiple installations of GDAL?

**Comment by Jinjin89** [comment]
I used conda doctor to diagnose my Conda environment, and it identified that some files, including liblzma, were missing. To resolve the issue, I force-reinstalled all the problematic packages using Conda, which successfully fixed most of the problems.

However, when I checked for liblzma using conda list, it showed that the package was already installed. Despite this, conda doctor continued to report liblzma as missing. Interestingly, after the reinstallation, I was able to install the terra package without any issues, even though the conda doctor warning about liblzma persisted.
